ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -217,6 +230,34 @@ public CancelHandle receiveInvocationRequest(@NotNull final InvocationRequest in
         return cancellationFlag::cancel;
     }
 
+    private static <A> boolean intersect(final Set<A> fst, final Set<A> snd) {
+        fst.retainAll(snd);
+        return !fst.isEmpty();
+    }
+
+    private static Set<String> typesOf(final Object o) {
```

Comment:
P.S.: I thought about List<Class<?>> as well but decided to avoid because that would introduce dependency between modules which is not otherwise required. A choice between bad and worse once again :)